src/test/librbd/test_mock_Journal.cc

```diff
@@ -186,6 +186,10 @@ OpenRequest<MockJournalImageCtx> *OpenRequest<MockJournalImageCtx>::s_instance =
 } // namespace librbd
 
 // template definitions
+
+#include "librbd/journal/PromoteRequest.cc"
+template class librbd::journal::PromoteRequest<librbd::MockJournalImageCtx>;
```

Comment:
Nit: this shouldn't be required (I have another PR to remove all of these) since it just leads to annoying compiler warnings about unused functions